File: src/main/java/com/github/novotnyr/mavenversion/ToggleShowMavenSettingsAction.java
package com.github.novotnyr.mavenversion;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class ToggleShowMavenSettingsAction extends ToggleAction {
    private Runnable onUpdateListener = () -> {};

    public ToggleShowMavenSettingsAction() {
        super("Show Maven Version in Modules");
    }

    @Override
    public boolean isSelected(@NotNull AnActionEvent e) {
        return PluginSettings.getInstance().isShowVersion();
    }

    @Override
    public void setSelected(@NotNull AnActionEvent e, boolean state) {
        PluginSettings.getInstance().setShowVersion(state);
        this.onUpdateListener.run();
    }

    public void setOnUpdateListener(Runnable onUpdateListener) {
        this.onUpdateListener = Objects.requireNonNull(onUpdateListener);
    }

    @Override
    public @NotNull ActionUpdateThread getActionUpdateThread() {
        return ActionUpdateThread.BGT;
    }
}

File: src/main/kotlin/com/github/novotnyr/mavenversion/ToggleShowMavenSettingsAction.kt
package com.github.novotnyr.mavenversion

import com.intellij.ide.projectView.ProjectView
import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project

class ToggleShowMavenSettingsAction : ToggleAction() {
    override fun isSelected(e: AnActionEvent): Boolean {
        return service<PluginSettings>().state.showVersion
    }

    override fun setSelected(e: AnActionEvent, state: Boolean) {
        service<PluginSettings>().state.showVersion = state
        e.project.projectView?.refresh()
    }

    override fun getActionUpdateThread() = ActionUpdateThread.BGT

    private val Project?.projectView: ProjectView?
        get() = this?.let { ProjectView.getInstance(it) }
}